File: Seafood Guide/Top 10/Table View Cell/DetailNewsViewController.m
//
//  DetailNewsViewController.m
//  Seafood Guide
//
//  Created by Jon Brown on 9/2/14.
//  Copyright (c) 2014 Jon Brown Designs. All rights reserved.
//

#import "DetailNewsViewController.h"
#import "NewsViewController.h"

@interface DetailNewsViewController ()

@end

@implementation DetailNewsViewController
@synthesize txtView,imgView,image,lblTitle,txtProject,myInt;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {

    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    #pragma mark - Button Style
    
    // R: 76 G: 76 B: 76
    UIColor *buttColor = [UIColor colorWithRed:(255.0 / 255.0) green:(255.0 / 255.0) blue:(255.0 / 255.0) alpha: 1];
    
    CALayer * layer = [customButton layer];
    [layer setMasksToBounds:YES];
    [layer setCornerRadius:4.0]; //when radius is 0, the border is a rectangle
    [layer setBorderWidth:1.0];
    [layer setBorderColor:[buttColor CGColor]];
    
    NSString* All = [self lblTitle];
    self.navigationItem.title = All;
}

-(void)viewDidAppear:(BOOL)animated
{

    [imgView setImage:image];
    [txtView setText:txtProject];
    [UIView setAnimationDuration:0.5f];
    [self.imgView setAlpha:1];
    [self.txtView setAlpha:1];
    
    [UIView commitAnimations];
}

-(void)viewDidDisappear:(BOOL)animated
{
    [self.imgView setImage:nil];
    [self.txtView setText:nil];
    [self.imgView setAlpha:0];
    [self.txtView setAlpha:0];
    

}



- (void)didMoveToParentViewController:(UIViewController *)parent
{
    if (![parent isEqual:self.parentViewController]) {
        
    }
}
- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}


-(UIButton*)createButtonWithFrame:(CGRect)frame andLabel:(NSString*)label
{
    
    UIButton *button = [[UIButton alloc] initWithFrame:frame];
    
    [button setTitleShadowColor:[UIColor blackColor] forState:UIControlStateNormal];
    button.titleLabel.shadowOffset = CGSizeMake(0, -1);
    button.titleLabel.font = [UIFont boldSystemFontOfSize:18];
    
    [button setTitle:label forState:UIControlStateNormal];
    [button setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    
    [[button layer] setBorderWidth:1.0f];
    [[button layer] setBorderColor:[UIColor whiteColor].CGColor];
    [[button layer] setMasksToBounds:YES];
    [[button layer] setCornerRadius:4.0]; //when radius is 0, the border is a rectangle
    [[button layer] setBorderWidth:1.0];
    
    return button;
}

-(void)closeView
{
    [self dismissViewControllerAnimated:YES completion:nil];
    [self.navigationController popViewControllerAnimated:YES];
    

    

}

@end

File: Seafood Guide/Top 10/Table View Cell/DetailNewsViewController.swift
//  Converted to Swift 5.1 by Swiftify v5.1.26565 - https://objectivec2swift.com/
//
//  DetailNewsViewController.swift
//  Seafood Guide
//
//  Created by Jon Brown on 9/2/14.
//  Copyright (c) 2014 Jon Brown Designs. All rights reserved.
//

import UIKit

@objcMembers
class DetailNewsViewController: UIViewController, UINavigationBarDelegate, UINavigationControllerDelegate {
    var str = ""
    var customButton: UIButton?
    var window: UIWindow?

    @IBOutlet var imgView: UIImageView!
    @IBOutlet var txtView: UITextView!
    var myInt = 0
    var image: UIImage?
    var lblTitle = ""
    var txtProject = ""

    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        navigationController?.navigationBar.barStyle = .default
        let appearance = UINavigationBarAppearance()
        appearance.titleTextAttributes = [.foregroundColor: UIColor(named: "Text Color")!]
        appearance.largeTitleTextAttributes = [.foregroundColor: UIColor(named: "Text Color")!]
        appearance.backgroundColor = UIColor(named: "Background")

        navigationItem.standardAppearance = appearance
        navigationItem.scrollEdgeAppearance = appearance
        self.navigationController?.navigationBar.tintColor = UIColor(named: "Text Color")!
        
// MARK: - Button Style

        // R: 76 G: 76 B: 76
        let buttColor = UIColor(red: 255.0 / 255.0, green: 255.0 / 255.0, blue: 255.0 / 255.0, alpha: 1)

        let layer = customButton?.layer
        layer?.masksToBounds = true
        layer?.cornerRadius = 4.0 //when radius is 0, the border is a rectangle
        layer?.borderWidth = 1.0
        layer?.borderColor = buttColor.cgColor

        let All = lblTitle
        navigationItem.title = All
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        imgView.image = image
        txtView.text = txtProject
        imgView.alpha = 1
        txtView.alpha = 1
    }

    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
        imgView.image = nil
        txtView.text = nil
        imgView.alpha = 0
        txtView.alpha = 0
    }

    override func didMove(toParent parent: UIViewController?) {
        if !(parent?.isEqual(self.parent) ?? false) {
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    func createButton(withFrame frame: CGRect, andLabel label: String?) -> UIButton? {

        let button = UIButton(frame: frame)

        button.setTitleShadowColor(UIColor.black, for: .normal)
        button.titleLabel?.shadowOffset = CGSize(width: 0, height: -1)
        button.titleLabel?.font = UIFont.boldSystemFont(ofSize: 18)

        button.setTitle(label, for: .normal)
        button.setTitleColor(UIColor.white, for: .normal)

        button.layer.borderWidth = 1.0
        button.layer.borderColor = UIColor.white.cgColor
        button.layer.masksToBounds = true
        button.layer.cornerRadius = 4.0 //when radius is 0, the border is a rectangle
        button.layer.borderWidth = 1.0

        return button
    }

    func closeView() {
        dismiss(animated: true)
        navigationController?.popViewController(animated: true)
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
}
